lib/publiccloud/provider.pm

```diff
@@ -705,6 +705,23 @@ sub terraform_param_tags
     return encode_json($tags);
 }
 
+=head2 terraform_query
+
+Query the terraform data structure.
+Input: jq-query string; empty = default query; '' = full json structure.
+
+=cut
+
+sub terraform_query {
+    my ($self, $qry) = @_;
```

Comment:
Why not use full variable name?  We're just saving 2 characters here.

```suggestion
    my ($self, $query) = @_;
```

Reply:
done